smallrye-reactive-messaging-mqtt/src/main/java/io/smallrye/reactive/messaging/mqtt/MqttSink.java

```diff
@@ -151,7 +162,7 @@ public void isReady(HealthReportBuilder builder) {
 
     public void isAlive(HealthReportBuilder builder) {
         if (healthEnabled)
-            builder.add(channel, isConnected());
+            builder.add(channel, !isDisconnected());
```

Comment:
Isn't having `isConnected` and `isDisconnected` methods confusing? 
With this line if `reference` is empty the liveness would be OK, is this intended?